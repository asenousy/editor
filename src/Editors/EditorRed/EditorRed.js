import { connect } from 'react-redux'
import Editor from '../Editor';
import styles from './EditorRed.module.css';
 
class EditorRed extends Editor {
  constructor(props) {
    super(props, styles);
  }
  static version() {
    return '1.0';
  }
}

export default connect(Editor.mapStateToProps, Editor.mapDispatchToProps)(EditorRed);
