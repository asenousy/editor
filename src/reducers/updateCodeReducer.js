export default (state = '', action) => {
    switch (action.type) {
        case 'UPDATE_CODE':
            return action.payload;
        default:
            return state;
    }
}