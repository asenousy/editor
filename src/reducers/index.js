import { combineReducers } from 'redux'
import updateCodeReducer from './updateCodeReducer'

const rootReducer = combineReducers({
  code: updateCodeReducer
})

export default rootReducer