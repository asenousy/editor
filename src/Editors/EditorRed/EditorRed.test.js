import React from 'react';
import renderer from 'react-test-renderer';
import EditorRed from './EditorRed';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import reducers from '../../reducers'
const store = createStore(reducers)

test('Editor renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><EditorRed /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});