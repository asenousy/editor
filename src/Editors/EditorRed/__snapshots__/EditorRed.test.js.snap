// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor renders correctly 1`] = `
<div>
  <h3>
    Version 
    1.0
    :
  </h3>
  <textarea
    onChange={[Function]}
    rows="10"
    value=""
  />
  <input
    hidden={true}
    onChange={[Function]}
    type="file"
  />
  <br />
  <a
    download={true}
    hidden={true}
  />
  <div>
    <button
      onClick={[Function]}
    >
      Upload
    </button>
    <button
      onClick={[Function]}
    >
      save
    </button>
  </div>
</div>
`;
